resources/virtual-machines/create-vm.js

```diff
@@ -125,39 +97,31 @@ const createVnet = async () => {
     },
     subnets: [{ name: subnetName, addressPrefix: "10.0.0.0/24" }],
   };
-  console.log("\n3.Creating vnet: " + vnetName);
-  return await networkClient.virtualNetworks.createOrUpdate(
-    resourceGroupName,
-    vnetName,
-    vnetParameters
-  );
-};
+  const resCreate = await networkClient.virtualNetworks.createOrUpdate(resourceGroupName,vnetName,vnetParameters);
+  return resCreate;
+}
 
-const getSubnetInfo = async () => {
+async function getSubnetInfo(){
   console.log("\nGetting subnet info for: " + subnetName);
-  return await networkClient.subnets.get(
-    resourceGroupName,
-    vnetName,
-    subnetName
-  );
-};
-const createPublicIP = async () => {
+  const getResult = await networkClient.subnet.get(resourceGroupName,vnetName,subnetName);
+  return getResult;
+}
+
+async function createPublicIP(){
+  console.log("\n4.Creating public IP: " + publicIPName);
   const publicIPParameters = {
     location: location,
     publicIPAllocationMethod: "Dynamic",
     dnsSettings: {
       domainNameLabel: domainNameLabel,
     },
   };
-  console.log("\n4.Creating public IP: " + publicIPName);
-  return await networkClient.publicIPAddresses.createOrUpdate(
-    resourceGroupName,
-    publicIPName,
-    publicIPParameters
-  );
-};
+  const resCreate = await networkClient.publicIPAddresses.createOrUpdate(resourceGroupName,publicIPName,publicIPParameters);
+  return resCreate;

```

Comment:
Is this createOrUpdate really exists in track2 SDK. I think it's a long running operation, the track1 createOrUpdate should map to beginCreateOrUpdateAndWait in track2.